### ISSUE TITLE
[LightGBM] [Fatal] Check failed: right_max_bin > 0

### ISSUE THREAD

**Issue by NickScottKortical** [microsoft/LightGBM#2480]
## Environment info

Operating System: Linux

CPU/GPU model:

C++/Python/R version: Python3.6

LightGBM version or commit hash:
lightgbm==2.2.3
## Error message

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .
[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .
[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .
[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Fatal] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .
terminate called after throwing an instance of 'std::runtime_error'

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

[LightGBM] [Warning] Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .

  what():  Check failed: right_max_bin > 0 at /__w/1/s/python-package/compile/src/io/bin.cpp, line 196 .
The documentation says that the constraints on max_bin are that it should be > 1, however it fails if you set it to 2, the minimum seems to be 3.
## Reproducible examples
Parameters set for lightgbm were:
    max_bin: 2

```import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder

from lightgbm import LGBMClassifier

train_df = pd.read_csv('/data_sets/titanic_train.csv')
test_df = pd.read_csv('/data_sets/titanic_test.csv')

embarked_encoder = LabelEncoder()
embarked_encoder.fit(train_df['Embarked'].fillna('Null'))

df_list = [train_df, test_df]

for df in df_list:
    df['Embarked'].fillna('Null', inplace=True)
    df['Embarked'] = embarked_encoder.transform(df['Embarked'])
    df.loc[df['Sex'] == 'female', 'Sex'] = 0
    df.loc[df['Sex'] == 'male', 'Sex'] = 1
    df['Sex'] = df['Sex'].astype('int8')
    df.drop(['Name', 'Ticket', 'Cabin'], axis=1, inplace=True)

y = train_df.pop('Survived')

X_train, X_test, y_train, y_test = train_test_split(train_df, y, test_size=0.2, random_state=42)
lgbm_params = {'max_bin': 2}
model = LGBMClassifier(**lgbm_params)
model.fit(X=X_train, y=y_train)
```
Dataset was the kaggle titanic dataset

## Steps to reproduce

1. Set the max bin parameter to 2
2. The error will be thrown during the fit() function

**Comment by StrikerRUS** [comment]
@NickScottKortical Thanks for reporting this! Please check `2.3.0`. It should be fixed there. Refer to #2299.

**Comment by NickScottKortical** [comment]
Ah great thanks, I only just picked up lightgbm last week and thought i was on the latest, but looks like theres a new version already :)

**Comment by StrikerRUS** [comment]
Yeah, we finally released `2.3.0` two days ago: https://github.com/microsoft/LightGBM/releases/tag/v2.3.0.
